fastparse/shared/src/main/scala/fastparse/core/Parsing.scala

```diff
@@ -370,7 +342,32 @@ trait Parser[+T, ElemType, Repr] extends ParserResults[T, ElemType] with Precede
             instrument: (Parser[_, _, _], Int, () => Parsed[_, ElemType]) => Unit = null)
            (implicit formatter: ElemTypeFormatter[ElemType])
             : Parsed[T, ElemType] = {
-    parseRec(new ParseCtx(input, 0, -1, this, index, instrument), index).toResult
+    parseRec(
```

Comment:
This and `parseIterator` below should probably just instantiate the relevant `ParserInput` and forward directly to `parseInput` to avoid all the repetition